pyvista/utilities/geometric_objects.py

```diff
@@ -726,21 +714,23 @@ def Pyramid(pointa, pointb, pointc, pointd, pointe):
     >>> pointc = [-1.0, -1.0, 1.0]
     >>> pointd = [1.0, -1.0, 1.0]
     >>> pointe = [0.0, 0.0, 0.0]
-    >>> pyramid = pyvista.Pyramid(pointa, pointb, pointc, pointd, pointe)
+    >>> pyramid = pyvista.Pyramid([pointa, pointb, pointc, pointd, pointe])
     >>> pyramid.plot() # doctest:+SKIP
     """
-    check_valid_vector(pointa, 'pointa')
-    check_valid_vector(pointb, 'pointb')
-    check_valid_vector(pointc, 'pointc')
-    check_valid_vector(pointd, 'pointd')
-    check_valid_vector(pointe, 'pointe')
+    assert(len(points) == 5)
+
+    check_valid_vector(points[0], 'points[0]')
+    check_valid_vector(points[1], 'points[1]')
+    check_valid_vector(points[2], 'points[2]')
+    check_valid_vector(points[3], 'points[3]')
+    check_valid_vector(points[4], 'points[4]')
 
-    points = vtk.vtkPoints()
-    points.InsertNextPoint(pointa)
-    points.InsertNextPoint(pointb)
-    points.InsertNextPoint(pointc)
-    points.InsertNextPoint(pointd)
-    points.InsertNextPoint(pointe)
+    pts = vtk.vtkPoints()
```

Comment:
We can simplify this by using `pyvista.utilities.helpers.vtk_points` to wrap the points array.